sway-core/src/ir_generation/convert.rs

```diff
@@ -159,6 +159,7 @@ fn convert_resolved_type(
             convert_resolved_typeid(type_engine, decl_engine, context, &ty.type_id, span)?
         }
         TypeInfo::Ref(_) => Type::get_uint64(context),
+        TypeInfo::Never => Type::get_never(context),
```

Comment:
Now that `convert_resolved_type` doesn't throw an error on diverging expressions we can simplify `compile_array_expr` in `function.rs`. It currently generates code for the first array element before creating the array, because creating the array requires the element type to be converted which wasn't possible when we used `Unknown` instead of `Never`.

With this change we don't need to use the first element as a guard against an illegal type conversion.